admin_guide/ipsec.adoc

```diff
@@ -0,0 +1,174 @@
+  pk12util -i certs.p12 -d sql:/etc/ipsec.d -W ""
+----
+
+[[ipsec-policy]]
+=== libreswan IPsec Policy
+
+Now that the necessary certificates have been imported into the `libreswan`
+certificate database, you can create a policy that uses them to secure
+communication between hosts in your cluster. The following configuration
+creates two `libreswan` connections.  The first encrypts traffic using the
+Openshift certificates, while the second creates exceptions to the encryption
+for cluster-external traffic.
+
+Place the following text into the file /etc/ipsec.d/openshift-cluster.conf,
+being sure to replace the `<certificate-nickname-here>` text with the 
+certificate nickname you saved above:
+
+----
+conn private
+	left=%defaultroute
+	leftid=%fromcert
+	# our certificate
+	leftcert="NSS Certificate DB:<certificate-nickname-here>"
+	right=%opportunisticgroup
+	rightid=%fromcert
+	# their certificate transmitted via IKE
+	rightca=%same
+	ikev2=insist
+	authby=rsasig
+	failureshunt=drop
+	negotiationshunt=hold
+	auto=ondemand
+
+conn clear
+	left=%defaultroute
+	right=%group
+	authby=never
+	type=passthrough
+	auto=route
+	priority=100
+----
+
+Now that the configuration has been defined, you need to tell `libreswan`
+what IP subnets and hosts to apply each policy to. This is done through policy
+files in /etc/ipsec.d/policies/ where each configured connection has a
+corresponding policy file. In our example above, we have two connections,
+`private` and `clear` and each will have a file in /etc/ipsec.d/policies.
+
+/etc/ipsec.d/policies/private should contain the IP subnet of your cluster,
+which your hosts receive IP addresses from. This, by default, will cause all
+communication between hosts in the 172.16.0.0/16 subnet to be encrypted if
+the remote host's client certificate authenticates against the local host's
+Certificate Authority certificate. If the remote host's certificate does not
+authenticate, all traffic between the two hosts it will be blocked.
+
+For example, if all your hosts are configured to use addresses in the
+172.16.0.0/16 address space, your `private` policy file would contain:
+
+----
+172.16.0.0/16
+----
+
+Next encryption between all hosts and the subnet gateway must be un-encrypted,
+to ensure that traffic can enter and exit the cluster.  To do this, add the
+gateway to the /etc/ipsec.d/policies/clear file like so:
+
+----
+172.16.0.1/32
+----
+
+Finally, restart the `libreswan` service to load the new configuration and
+policies, and begin encrypting:
+
+----
+systemctl restart ipsec
+----
+
+== Troubleshooting
+When authentication cannot be completed between two hosts, you will not even
+be able to ping between them as all IP traffic will be rejected. If the `clear`
+policy is not configured correctly, you will also not be able to SSH to the host
+from another host in the cluster.
+
+The most common failure is certificates that do not verify against a common
+Certificate Authority certificate. To verify this:
```

Comment:
It seems unlikely that this would happen in an OpenShift cluster, since the certificates are being created by automated processes; if people were ending up with node certs that used the wrong CA, then that would cause https failures even if they weren't using ipsec.